bigflow/bigquery/dataset_manager.py

```diff
@@ -309,9 +311,20 @@ def create_table(self, create_query):
         job = self.bigquery_client.query(
             create_query,
             job_config=job_config)
-        return job.result()
+        job_result = job.result()
```

Comment:
I'm wondering if that is the best place to apply labels because not all tables must be created using `create_table`. It's also confusing because you might think that if you declare labels in the dataset configuration, labels will be applied "no matter what". I suggest alternative solutions:

- creating table labels in the `create_dataset_manager` function
- adding an explicit, optional argument to the `create_table` method
- creating new method on dataset manager, called `upsert_table_labels`

also, @anjensan what do you think?

Reply:
I was thinking about all solutions you've mentioned but I implemented labels in this way because:
> creating table labels in the create_dataset_manager function
If the table does not exist yet then we can't add a label.

>adding an explicit, optional argument to the create_table method
User must provide labels for every call of create_table which is uncomfortable 

>creating new method on dataset manager, called upsert_table_labels
It adds another separate step and users need to remember about adding it.

Maybe adding a prefix to param names will be enough?

` tables_labels, dataset_labels` -> ` create_table_labels, create_dataset_labels`

Reply:
"If the table does not exist yet then we can't add a label." -> is it a problem?

Reply:
> "If the table does not exist yet then we can't add a label." -> is it a problem?

Yes, because then we will need to define labels during creating a table so we will be touching labels in two places

Reply:
can't we just do something like this in `create_dataset_manager`?:

```python
if table_exists:
    upsert_labels()
```

Reply:
(and don't do anything in `create_table`)

Reply:
But if a table does not exist then we have to pass labels in `create_table` method. So instead of passing labels only to `create_dataset_manager` we have to pass in two places.

Reply:
I think we can just have the logic in the `create_dataset_manager` method and assume that if that's the first run, before `create_table`, there will be no labels until the second run

Reply:
Ok, will change to that.

Reply:
See now